### PR TITLE
Entry point on github actions

### PR DESCRIPTION
**Pull request by alorma**
From https://github.com/AdevintaSpain/Barista/issues/333

Add Github actions CI to run instrumentation tests on  every push on master and PR

### CODE REVIEW

File: sample/src/androidTest/java/com/schibsted/spain/barista/sample/MenuClickTest.java [hunk 1]
```diff
@@ -46,18 +47,21 @@ public void iconMenuClick_byTitle() {
   }
 
   @Test
+  @Ignore
```
Comment (alorma) [inline sample/src/androidTest/java/com/schibsted/spain/barista/sample/MenuClickTest.java]: Those are flaky tests on CI, we must try to check why...

Comment (rocboronat) [reply sample/src/androidTest/java/com/schibsted/spain/barista/sample/MenuClickTest.java]: I think they pass in the local machine...

Comment (alorma) [reply sample/src/androidTest/java/com/schibsted/spain/barista/sample/MenuClickTest.java]: Yes, that's why I marked them as flaky, maybe we can merge this PR and try to fix the tests?

Comment (rocboronat) [reply sample/src/androidTest/java/com/schibsted/spain/barista/sample/MenuClickTest.java]: You marked them as `@Ignore`, not flaky.

Perwaps we could add a message that explains that it doesn't pass on CI but it does in the local machine? Just to help to ourselves of the future.

Comment (alorma) [reply sample/src/androidTest/java/com/schibsted/spain/barista/sample/MenuClickTest.java]: Let's see if `@FlakyTest` works

Comment (Sloy) [reply sample/src/androidTest/java/com/schibsted/spain/barista/sample/MenuClickTest.java]: Why do they fail in CI? Do we have any idea?
Maybe we could try running it locally with the exact same emulator as CI, sometimes that gives it away 🤷‍♂

**Comment by alorma** [comment]
I've updated the emulator plugin version to 2.

It allows to create a matrix of emulators, cool!

There are still failing tests...

I think...

Could we merge this, and let's fix tests later? This way at least we can track failing tests, as it happened on #350 , we didn't know that some tests were failing

**Comment by rocboronat** [comment]
But we're going to have tests that pass on local but not in the CI... so we'll ignore them I think 😅

**Comment by alorma** [comment]
Not really.

Isn't it better to have CI with failing tests that no CI with no failing tests?

Right now on travis we are not checking that...

**Comment by alorma** [comment]
I reproduced the failing tests on local.

**Comment by rocboronat** [comment]
`Isn't it better to have CI with failing tests that no CI with no failing tests?`

Hum, that's ok if you had green builds before, but if you never had the suite passing, I don't see the value but having red builds on production, which is eye-painful hehe... but it's just a personal decision I think. I understand the CI as a fast way to see errors before sending things to production. If it says that everything is incorrect but actually all is working as expected, I don't know...

**Comment by Sloy** [comment]
I agree with @rocboronat. Having a red build all the time doesn't help. You get used to it and ignore it :(

**Comment by rocboronat** [comment]
@alorma is very near to achieve having the tests passing here. Can't wait to see them working! ❤️

**Comment by alorma** [comment]
Look @Sloy @rocboronat it's green!

**Comment by rocboronat** [comment]
Cool! If you think it's stable enough, feel free to merge. If it doesn't behave as we expect, we always have time to disable it. Let's try it in the real world!

Thanks for working on it, Bernat! 😊 For sure a lot of people will be "influenced" by this PR 😄

**Pull request merged**